### PR TITLE
add debug message in interop test api_listener

### PR DESCRIPTION
**Pull request by YifeiZhuang**
b/216278078

cc. @sergiitk

### CODE REVIEW

**Review by lidizheng — APPROVED** [review]

File: tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py [hunk 1]
```diff
@@ -37,6 +37,8 @@
 from framework.test_app import client_app
 from framework.test_app import server_app
 
+from framework.rpc.grpc_csds import ClientConfig
```
Comment (lidizheng) [inline tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py:40]: You may want to use a type alias as below

File: tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py [hunk 2]
```diff
@@ -364,12 +365,17 @@ def getRouteConfigVersion(test_client: XdsTestClient) -> Optional[str]:
                 logger.info('found routing config version: %s',
                             route_config_version)
                 break
+        if not route_config_version:
+            logger.info('no route config version found.')
         return route_config_version
```
Comment (lidizheng) [inline tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py:370]: Will it return None? If not, let's set the return type to `str`.

Comment (YifeiZhuang) [reply tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py:370]: I believed it returned None in the CI, so I added this line to confirm.

**Comment by lidizheng** [comment]
PTAL at sanity checks

**Comment by lidizheng** [comment]
The debug info about xDS config is still quite valuable if we can print them.

**Pull request closed without merging**